presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -739,7 +745,10 @@ public static Signature getMagicLiteralFunctionSignature(Type type)
     {
         TypeSignature argumentType = typeForMagicLiteral(type).getTypeSignature();
 
-        return new Signature(MAGIC_LITERAL_FUNCTION_PREFIX + type.getTypeSignature(),
+        // The RowType have <> that are not allowed in java variable names, so we replace to {}. We can remove this
+        // hack when the type ROW doesn't have them anymore (be ROW(field type+) instead of ROW<type+>(field+))
+        String typeName = type.getTypeSignature().toString().replace('<', '{').replace('>', '}');
+        return new Signature(MAGIC_LITERAL_FUNCTION_PREFIX + typeName,
```

Comment:
ditto.
I can wait for https://github.com/prestodb/presto/pull/5043 to remove this and then land.